sway-lib-std/src/flags.sw

```diff
@@ -30,3 +89,196 @@ pub fn enable_panic_on_overflow() {
         flag flag_val;
     }
 }
+
+/// Allows overflowing operations to occur without a FuelVM panic.
+/// More suitable than `disable_panic_on_overflow` for use in functions that are not the entry point of a program.
+///
+/// > **_WARNING:_**
+/// >
+/// > Don't forget to call `set_flags` after performing the operations for which you disabled the default `panic-on-overflow` behavior in the first place!
+///
+/// ### Examples
+///
+/// ```sway
+/// use std::flags::{disable_panic_on_overflow_preserving, set_flags};
+///
+/// fn foo() {
+///     let prior_flags = disable_panic_on_overflow_preserving();
+///      
+///     let bar = u64::max() + 1;
+///
+///     set_flags(prior_flags);
+/// }
+/// ```
+pub fn disable_panic_on_overflow_preserving() -> u64 {
+    let prior_flags = flags();
+
+    // Mask second bit, which is `F_WRAPPING`.
+    let mask = 0b00000000_00000000_00000000_00000000_00000000_00000000_00000000_00000010;
+
+    // Get the current value of the flags register and mask it, setting the
+    // masked bit. Flags are inverted, so set = off.
+    let flag_val = prior_flags | mask;
+    asm(flag_val: flag_val) {
+        flag flag_val;
+    }
+
+    prior_flags
+}
+
+/// Allows unsafe math operations to occur without a FuelVM panic.
+/// Sets the `$err` register to `true` whenever unsafe math occurs.
+///
+/// > **_WARNING:_**
+/// >
+/// > Don't forget to call `enable_panic_on_unsafe_math` after performing the operations for which you disabled the default `panic-on-unsafe-math` behavior in the first place!
+///
+/// ### Examples
+///
+/// ```sway
+/// use std::{assert::assert, flags::{disable_panic_on_unsafe_math, enable_panic_on_unsafe_math}, registers::error};
+///
+/// fn main() {
+///     disable_panic_on_unsafe_math();
+///      
+///     let bar = 1 / 0; // Division by zero is considered unsafe math.
```

Comment:
```suggestion
///     // Division by zero is considered unsafe math.
///     let bar = 1 / 0; 
```
This isn't really necessary but bothered me sorry. Same thing is done one line bellow..